public/app/percona/dbaas/components/PMMServerURLWarning/PMMServerUrlWarning.hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { useState, useEffect, useMemo } from 'react';
+import { SettingsService } from '../../../settings/Settings.service';
+import { Settings } from '../../../settings/Settings.types';
+import { logger } from '@percona/platform-core';
+
+export const usePMMServerWarning = (): boolean => {
+  const [settingsLoading, setSettingsLoading] = useState(true);
+  const [settings, setSettings] = useState<Settings>();
+
+  const showMonitoringWarning = useMemo(() => settingsLoading || !settings?.publicAddress, [settings, settingsLoading]);
+
+  useEffect(() => {
+    const getSettings = async () => {
+      try {
+        setSettingsLoading(true);
+        const settings = await SettingsService.getSettings();
+        setSettings(settings);
+      } catch (e) {
+        logger.error(e);
+      } finally {
+        setSettingsLoading(false);
+      }
+    };
+
+    getSettings();
+  }, []);
+
+  return showMonitoringWarning;
+};
```

Comment:
Warning: this is gonna be deeply affected by this PR: https://github.com/percona-platform/grafana/pull/340
We'll no longer call the API, to get the settings, but rely on the store instead. This said, I think this hook won't be necessary.

Reply:
thanks, I understand, I'm going to study this PR

Reply:
fixed